storage/client_test.go

```diff
@@ -276,7 +277,13 @@ func (m *multiTestContext) rpcSend(_ rpc.Options, _ string, addrs []net.Addr,
 		if stErr != nil {
 			m.t.Fatal(stErr)
 		}
-		br, pErr = m.senders[nodeID-1].Send(context.Background(), ba)
+		nodeIndex := nodeID - 1
+		if m.stores[nodeIndex] == nil {
+			pErr = &roachpb.Error{}
+			pErr.SetGoError(rpc.NewSendError("store is stopped", true))
+			continue
```

Comment:
Yes, this is intended. `pErr` exists outside the loop. We return permanent errors but for retryable errors we continue through the loop until we've exhausted all the stores (at which point we return the last such error)